src/parquet/column/serialized-page.cc

```diff
@@ -45,23 +45,36 @@ SerializedPageReader::SerializedPageReader(std::unique_ptr<InputStream> stream,
   }
 }
 
-// TODO(wesm): this may differ from file to file
-static constexpr int DATA_PAGE_SIZE = 64 * 1024;
+static constexpr int PAGE_SIZE = 64 * 1024;
+static constexpr int MAX_PAGE_SIZE = 16 * 1024 * 1024;
```

Comment:
16 MB is too small (hence making this configurable); the page statistics can be arbitrarily large. Impala has observed pathological cases with page headers exceeding 2 GB.